www/docs/alert/index.php

```diff
@@ -341,6 +341,38 @@ function display_search_form ( $alert, $details = array(), $errors = array() ) {
             echo '<em class="error">You have used a comma in your search term &ndash; are you sure this is what you want?
 You cannot sign up to multiple search terms using a comma &ndash; either use OR, or fill in this form multiple times.</em>';
         }
+        if ( preg_match('#([A-Z]{1,2}\d+ ?\d[A-Z]{2})#i', $alertsearch, $m) && strlen($alertsearch) > strlen($m[1]) && validate_postcode($m[1]) ) {
+            $scottish_text = '';
+            $mp_display_text = '';
+            if (postcode_is_scottish($m[1])) {
+                $mp_display_text = 'your MP, ';
+                $scottish_text = ' or MSP';
+            }
+            echo '<em class="error">You have used a postcode and something else in your search term &ndash; are you sure this is what you want?
+                  You will only get an alert if all of these are mentioned in the same debate. Did you mean to get alerts for when your MP'
+                  . $scottish_text . ' mentions something instead? If so click subscribe below.</em></li>';
+
+            $MEMBER = new MEMBER(array('postcode'=>$m[1]));
+            // move the postcode to the front just to be tidy
+            $tidy_alertsearch= $m[1] . " " . trim(str_replace("$m[1]", "", $alertsearch));
+            $alertsearch_display = str_replace("$m[1] ", "", $tidy_alertsearch);
+            $alertsearch = str_replace("$m[1]", "speaker:" . $MEMBER->person_id, $tidy_alertsearch);
+            echo "<li>";
+            echo $form_start . '<input type="hidden" name="keyword" value="' . _htmlspecialchars($alertsearch) . '">';
+            echo 'Mentions of [';
+            echo _htmlspecialchars($alertsearch_display) . '] by ' . $mp_display_text . $MEMBER->full_name();
+            echo ' <input type="submit" value="Subscribe"></form>';
+            if ( $scottish_text ) {
+                $MEMBER = new MEMBER(array('postcode'=>$m[1], 'house' => 4));
```

Comment:
Ah, actually it won't work even if both are present; the postcode lookup of MEMBER only appears to work for MPs (it only calls `postcode_to_constituency` which returns the MP constituency name, not the perhaps different MSP one, e.g. EH11BB East/Central). It looks like mp/index.php and postcode/index.php both do the same sort of thing (they have the same fix me comment also) for getting MSPs/MLAs for a postcode, and something different would have to be done here, I'm afraid.